### DISCUSSION TITLE
命令式调用funcilp似乎很笨，有考虑做api吗？

### DISCUSSION THREAD

**Discussion by 851543** [modelscope/FunClip#62] [Q&A] [answered]
真的！！！

**Comment by 851543** [comment]
命令式只有剪辑功能吗？没有剪辑+字幕？

**Comment by LauraGPT** [comment, marked as answer]
现在已经有可直接调用的 API，不过需要区分“公共体验 API”和“本地完整版”。

### 1. 公共 Hugging Face Space API

当前 [FunClip Space](https://huggingface.co/spaces/FunAudioLLM/FunClip) 暴露了三个命名端点：

- `/transcribe_video`：转写视频并返回句级时间戳文本
- `/build_selector`：读取本次转写产生的可选片段
- `/clip_video`：按选择的片段剪辑并返回视频

这个公共 Space 当前固定使用中文 Paraformer；多语种模型、说话人识别和 LLM 辅助剪辑请部署本地完整版。

可以用同一个 `gradio_client.Client` 实例按顺序调用：

```python
from gradio_client import Client, handle_file

client = Client("FunAudioLLM/FunClip")
video = {
    "video": handle_file("input.mp4"),
    "subtitles": None,
}

# 1. 转写；同时在当前 Client 会话中保存句级时间戳状态
transcript = client.predict(video, api_name="/transcribe_video")

# 2. 取得片段选项。Gradio 返回的是 update 字典，choices 中每项为 [label, value]
selector = client.predict(api_name="/build_selector")
choices = [item[1] for item in selector["choices"]]

# 3. 示例：剪辑第一个句级片段
clip, info = client.predict(
    video,
    choices[:1],
    api_name="/clip_video",
)

print(transcript)
print(clip["video"])
print(info)
```

安装客户端：

```bash
pip install -U gradio_client
```

接口参数也可以在 [Space API 页面](https://funaudiollm-funclip.hf.space/?view=api) 查看。必须复用同一个 `Client` 并保持上述顺序，因为句级时间戳状态属于该 Gradio 会话。

我刚用 20.04 秒的官方云栖大会样例重新验证：接口返回 9 个句级片段，选择第一段后生成了 1.50 秒、84,824 字节的 H.264/AAC MP4。

### 2. 字幕能力的准确边界

公共 Space API 当前返回带时间戳的文本和剪辑视频，输出视频的 `subtitles` 为 `None`，也没有字幕流，因此它目前不是“剪辑并烧录字幕”API。

本地完整版已经支持字幕：

- CLI `--stage 1` 会生成 `output/total.srt`。
- CLI `--stage 2 --output_file output/res.mp4` 会同时生成 `output/res.mp4` 和 `output/res.srt`，这里的字幕是独立 SRT 文件，不会烧录进视频画面。
- 运行 `python funclip/launch.py` 后，页面中的“裁剪+字幕”会把字幕烧录到视频；这条路径需要安装 FFmpeg 和 ImageMagick。

所以目前的答案是：**有 API，也有剪辑和 SRT；公共 API 暂不烧录字幕，本地 UI 支持烧录字幕。** 公共 Space 使用共享 CPU，适合体验和轻量集成；生产服务建议在自己的机器上部署。
